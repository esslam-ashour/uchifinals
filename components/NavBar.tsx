import localFont from "next/font/local";
import Link from "next/link";

const fettefraktur = localFont({
  src: "../public/fonts/fettefraktur.ttf"
});

export default function NavBar() {
  return (
    <header className="text-white py-4">
      <div className="container mx-auto flex flex-col md:flex-row items-center justify-between px-4">
        <Link href="/">
          <div className="flex flex-col items-center md:items-start">
            <h1 className={`${fettefraktur.className} text-3xl md:text-5xl`}>
              uchifinals
            </h1>
            <p className="text-xs md:text-sm mt-1 text-center md:text-left">
              find your uchicago final exam schedule fast
            </p>
            <p className="text-xs md:text-sm mt-1 text-center md:text-left">
              current quarter: <span className="font-bold">autumn 2024</span>
            </p>
          </div>
        </Link>
        <nav className="mt-4 md:mt-0">
          <ul className="flex flex-col md:flex-row gap-4 md:gap-6 text-xs md:text-base font-bold items-center">
            <li className="text-center">
              Made by Esslam Ashour, class of 2027
            </li>
            <li>
              <a href="mailto:esslamashour@uchicago.edu" className="text-[#800000] bg-white p-2 md:p-3 rounded-3xl font-normal hover:underline text-center block">
                Contact
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </header>
  );
}
