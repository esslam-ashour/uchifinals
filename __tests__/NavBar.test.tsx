import { render } from '@testing-library/react';
import NavBar from '@/components/NavBar';

describe('NavBar', () => {
  it('renders correctly', () => {
    const { container } = render(<NavBar />);
    expect(container).toMatchSnapshot();
  });
});

