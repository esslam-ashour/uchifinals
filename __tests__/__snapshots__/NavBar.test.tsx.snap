// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
<div>
  <header
    class="text-white py-2 md:py-4"
  >
    <div
      class="container mx-auto flex flex-col md:flex-row items-center justify-between px-2 md:px-4"
    >
      <a
        href="/"
      >
        <div
          class="flex flex-col items-center md:items-start"
        >
          <h1
            class="className text-2xl md:text-5xl"
          >
            uchifinals
          </h1>
          <p
            class="text-xs md:text-sm mt-1 text-center md:text-left"
          >
            find your uchicago final exam schedule fast
          </p>
          <p
            class="text-xs md:text-sm mt-1 text-center md:text-left"
          >
            current quarter: 
            <span
              class="font-bold"
            >
              autumn 2024
            </span>
          </p>
        </div>
      </a>
      <nav
        class="mt-2 md:mt-0"
      >
        <ul
          class="flex flex-col md:flex-row gap-2 md:gap-6 text-xs md:text-base font-bold items-center"
        >
          <li
            class="text-center"
          >
            Made by Esslam Ashour, class of 2027
          </li>
          <li>
            <a
              class="text-[#800000] bg-white p-1 md:p-3 rounded-3xl font-normal hover:underline text-center block"
              href="mailto:esslamashour@uchicago.edu"
            >
              Contact
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </header>
</div>
`;
